libcaf_core/caf/flow/observable_decl.hpp

```diff
@@ -173,6 +173,12 @@ class observable {
     return reduce(T{}, [](T x, T y) { return x + y; });
   }
 
+  /// Adds the output of fn() to the beginning of current observable.
+  template <class F>
+  auto start_with(F fn) {
+    return ctx()->make_observable().just(fn()).concat(*this);
```

Comment:
I don't see much upside for taking a function object as argument. We call it immediately anyways, so I'd rather take the value directly. Furthermore, if we take a value, we can dispatch on `if constexpr (is_observable_v<T>)` to allow users to call the function with a value or another observable. In the later case, we can just call `std::move(value).concat(*this)`.